cloudsql/postgres/database-sql/connect_connector_iam_authn.go

```diff
@@ -47,7 +47,11 @@ func connectWithConnectorIAMAuthN() (*sql.DB, error) {
 		usePrivate             = os.Getenv("PRIVATE_IP")
 	)
 
-	d, err := cloudsqlconn.NewDialer(context.Background(), cloudsqlconn.WithIAMAuthN())
+	// WithLazyRefresh() Option is used to perform refresh
+	// when needed, rather than on a scheduled interval.
+	// this is recommended for serverless environments to
+	// avoid background refreshes from throttling CPU.
+	d, err := cloudsqlconn.NewDialer(context.Background(), cloudsqlconn.WithIAMAuthN(), cloudsqlconn.WithLazyRefresh())
```

Comment:
Let's format this across several lines so that in the docs a user does not need to scroll sideways...
```go
d, err := cloudsqlconn.NewDialer(
    context.Background(),
    cloudsqlconn.WithIAMAuthN(),
    cloudsqlconn.WithLazyRefresh(),
)
```